lib/features/email_recovery/presentation/model/email_recovery_field.dart

```diff
@@ -40,16 +40,20 @@ enum EmailRecoveryField {
     }
   }
 
-  List<EmailRecoveryTimeType> getSupportedTimeTypes() {
+  List<EmailRecoveryTimeType> getSupportedTimeTypes(DateTime restorationHorizon) {
     switch (this) {
       case EmailRecoveryField.deletionDate:
-        return [
-          EmailRecoveryTimeType.last1Year,
+        final supportedTypes = [
           EmailRecoveryTimeType.last7Days,
+          EmailRecoveryTimeType.last15Days,
           EmailRecoveryTimeType.last30Days,
           EmailRecoveryTimeType.last6Months,
-          EmailRecoveryTimeType.customRange,
-        ];
+          EmailRecoveryTimeType.last1Year,
+        ].where((type) => restorationHorizon
+                                .subtract(const Duration(seconds: 2)) // to allow "15 days" if restorationHorizon is exactly 15
```

Comment:
Why subtract `2s` and not some other number?

Reply:
i figured that it was small enough to not significantly interfere with the horizon (smallest reasonable horizon would be around 1 minute, so 2 seconds does not change much), while making sure the desired option is considered valid.
but it is not an absolute choice, it can be changed